src/components/FormSubmit/index.js

```diff
@@ -0,0 +1,57 @@
+import React from 'react';
+import lodashGet from 'lodash/get';
+import {View} from 'react-native';
+import * as formSubmitPropTypes from './formSubmitPropTypes';
+
+// This is a wrapper component to handle the ENTER key press, and submit the form.
+class FormSubmit extends React.Component {
+    constructor(props) {
+        super(props);
+
+        this.submitForm = this.submitForm.bind(this);
+    }
+
+    /**
+     * Submits the form when ENTER is pressed on form element.
+     * @param {SyntheticEvent} event
+     */
+
+    submitForm(event) {
+        // ENTER is pressed with modifier key, do not submit the form
+        if (event.shiftKey || event.key !== 'Enter') {
+            return;
+        }
+
+        const tagName = lodashGet(event, 'target.tagName', '');
+
+        // ENTER is pressed on INPUT or SELECT element, submit the form
+        if (tagName === 'INPUT' || tagName === 'SELECT') {
+            this.props.onSubmit();
+            return;
+        }
+
+        // We usually do not submit the form on TEXTAREA element, when ENTER is pressed.
+        // As there is a prop on react native `blurOnSubmit`, we indeed want to submit it here.
+        // Element dataset is used here to store `blurOnSubmit` prop.
```

Comment:
```suggestion
        // Pressing Enter on TEXTAREA element adds a new line. When `dataset.blurOnSubmit` prop is passed, call submit.
```

Reply:
I don't think the prop name suites the case. We want to submit the form upon entering but blurOnSubmit has a different purpose. IMO, it should be some other name and the value should be bool. Could you please remind me if this was already discussed somewhere?

Reply:
If this prop is only applicable to TextInput then we can create a new one directly on our TextInput Component.

Reply:
Ok, let's do this.

Reply:
```
// Pressing Enter on TEXTAREA element adds a new line. When `dataset.submitOnEnter` prop is passed, call submit.
```

Is it good?

Reply:
FormSubmit `Enter` key handler does not have access to direct props. `dataset.submitOnEnter` is used to indicate that pressing Enter on this input should call the submit callback.

Reply:
Thanks!